internal/metamorph/store/postgresql/postgres_helpers.go

```diff
@@ -19,3 +35,194 @@ func mergeUnique(arr1, arr2 []string) []string {
 
 	return uniqueSlice
 }
+
+func getStoreDataFromRows(rows *sql.Rows) ([]*store.StoreData, error) {
+	var storeData []*store.StoreData
+
+	for rows.Next() {
+		data := &store.StoreData{}
+
+		var announcedAt sql.NullTime
+		var minedAt sql.NullTime
+		var status sql.NullInt32
+
+		var txHash []byte
+		var blockHeight sql.NullInt64
+		var blockHash []byte
+
+		var callbackUrl sql.NullString
+		var callbackToken sql.NullString
+		var rejectReason sql.NullString
+		var competingTxs string
+		var merklePath sql.NullString
+		var retries sql.NullInt32
+
+		err := rows.Scan(
+			&data.StoredAt,
+			&announcedAt,
+			&minedAt,
+			&txHash,
+			&status,
+			&blockHeight,
+			&blockHash,
+			&callbackUrl,
+			&callbackToken,
+			&data.FullStatusUpdates,
+			&rejectReason,
+			&competingTxs,
+			&data.RawTx,
+			&data.LockedBy,
+			&merklePath,
+			&retries,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(txHash) > 0 {
+			data.Hash, err = chainhash.NewHash(txHash)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if len(blockHash) > 0 {
+			data.BlockHash, err = chainhash.NewHash(blockHash)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if announcedAt.Valid {
+			data.AnnouncedAt = announcedAt.Time.UTC()
+		}
+
+		if minedAt.Valid {
+			data.MinedAt = minedAt.Time.UTC()
+		}
+
+		if status.Valid {
+			data.Status = metamorph_api.Status(status.Int32)
+		}
+
+		if blockHeight.Valid {
+			data.BlockHeight = uint64(blockHeight.Int64)
+		}
+
+		if callbackUrl.Valid {
+			data.CallbackUrl = callbackUrl.String
+		}
+
+		if callbackToken.Valid {
+			data.CallbackToken = callbackToken.String
+		}
+
+		if rejectReason.Valid {
+			data.RejectReason = rejectReason.String
+		}
```

Comment:
I think there is no reason to check if the field is `valid` here

Reply:
I'll remove that in the next PR